explorer/src/Pos/Explorer/Web/Server.hs

```diff
@@ -711,8 +715,31 @@ topsortTxsOrFail f =
 
 -- | Deserialize Cardano or RSCoin address and convert it to Cardano address.
 -- Throw exception on failure.
-cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr@(CAddress rawAddrText) =
+-- cAddrToAddr :: MonadThrow m => CAddress -> m Address
+-- cAddrToAddr cAddr@(CAddress rawAddrText) =
+--     -- Try decoding address as base64. If both decoders succeed,
+--     -- the output of the first one is returned
+--     let mDecodedBase64 =
+--             rightToMaybe (B64.decode rawAddrText) <|>
+--             rightToMaybe (B64.decodeUrl rawAddrText)
+--     in case mDecodedBase64 of
+--         Just addr -> do
+--             -- cAddr is in RSCoin address format, converting to equivalent Cardano address
+--             -- Originally taken from:
+--             -- * cardano-sl/tools/src/keygen/Avvm.hs
+--             -- * cardano-sl/tools/src/addr-convert/Main.hs
+--             unless (BS.length addr == 32) $
+--                 throwM badAddressLength
+--             pure $ makeRedeemAddress $ redeemPkBuild addr
+--         Nothing ->
+--             -- cAddr is in Cardano address format
+--             either badCardanoAddress pure (fromCAddress cAddr)
+--   where
+--     badAddressLength = Internal "Address length is not equal to 32, can't be redeeming pk"
+--     badCardanoAddress = const $ throwM $ Internal "Invalid Cardano address!"
```

Comment:
In particular, we're removing a bunch of checks here. What is the idea behind this?

Also, don't leave commented-out code like this.

Reply:
The checks were removed since only RSCoin addresses have that invariant, other addresses don't have it.
I commented out the code since I wanted to leave a function people can run in a test to see why it fails and left the original commented so they can see the difference.
It's not like I usually leave commented out code around.

Reply:
Comment from @gromakovsky:

Decoded RSCoin (AVVM) address must have exactly 32 bytes, but Cardano address is different and its length is not fixed. I don't know how `cAddrToAddr` is used, but it makes sense only for RSCoin addresses.